src/mutation-observer.js

```diff
@@ -63,62 +63,26 @@ const clearNode = node => {
 	}
 }
 
-const reapplyFocus = mutationList => {
-	// check if any nodes had focus
-	const hasFocus = node => node[TRAM_TAG_FOCUS]
-
-	const removedNodes = mutationList
-		.flatMap(mutation => [...mutation.removedNodes])
-		.flatMap(node => [...node.querySelectorAll('*')])
-
-	const elementIndexWithFocus = removedNodes.findIndex(hasFocus)
-	const removedElementWithFocus = removedNodes[elementIndexWithFocus]
-
-	// if we couldn't find any, we can leave now
-	if (elementIndexWithFocus === -1) { return }
-
-	// check if the number of elements matches (if it does, we can probably focus the same index)
-	const addedNodes = mutationList
-		.flatMap(mutation => [...mutation.addedNodes])
-		.flatMap(node => [...(node.querySelectorAll ? node.querySelectorAll('*') : [])])
-
-	const sameNumberOfElements = addedNodes.length === removedNodes.length
-
-	// if the number of elements changed, we can't confidently re-apply focus
-	if (!sameNumberOfElements) { return }
-
-	// if we found the same number of elements, let's re-apply focus to the same child
-	addedNodes[elementIndexWithFocus].focus()
-
-	// give it the attribute in case it is edited again
-	addedNodes[elementIndexWithFocus][TRAM_TAG_FOCUS] = true
-
-	// also try to set the selection, if there is a selection for this element
-	const hasSelectionStart = removedElementWithFocus.selectionStart !== null && removedElementWithFocus.selectionStart !== undefined
-	if (hasSelectionStart) {
-		addedNodes[elementIndexWithFocus].setSelectionRange(
-			removedElementWithFocus.selectionStart,
-			removedElementWithFocus.selectionEnd,
-			removedElementWithFocus.selectionDirection
-		)
-	}
+// function to get the children (as a list) of the node passed in
+const childrenNodes = node => {
+	const children = node.querySelectorAll ? node.querySelectorAll('*') : []
+	return [...children]
 }
 
 const setupMutationObserver = setup(() => new MutationObserver(mutationList => {
 	// cleanup orphaned nodes that are no longer on the DOM
-	mutationList
+	const removedNodes = mutationList
 		.flatMap(mutation => [...mutation.removedNodes])
-		.flatMap(node => [...node.querySelectorAll('*')])
-		.forEach(clearNode)
+		.flatMap(childrenNodes)
+
+	removedNodes.forEach(clearNode)
 
 	// call new effects on any new nodes
-	mutationList
+	const newNodes = mutationList
 		.flatMap(mutation => [...mutation.addedNodes])
-		.flatMap(node => [...(node.querySelectorAll ? node.querySelectorAll('*') : [])])
-		.forEach(processEffects)
+		.flatMap(childrenNodes)
 
-	// if any node had focus, get that element and re-add focus to the new node
-	reapplyFocus(mutationList)
+	newNodes.forEach(processEffects)
```

Comment:
this logic didn't actually change, we just made it more clear by making a new `childrenNodes()` function (since that's all this logic interacts with).

Reply:
To be honest, I'm not actually sure why this only works with just children nodes, or if that's actually intentional. I feel like this might bite us in the butt later, but it doesn't work with the node and children, and this does work, so eh :man_shrugging: